config/v2/validation.go

```diff
@@ -91,6 +87,41 @@ func ValidateAWSProvider(p *AWSProvider, component string) error {
 	return errs
 }
 
+// ValidateBackend will check the resolved configuration for a backend and return any errors it
+// finds
+func ValidateBackend(backend *Backend, component string) error {
```

Comment:
does this need validation to make sure only one backend (and corresponding fields) is enabled?

Reply:
Yeah it does. I had been thinking it was impossible to get to that state, but that is only true at the planning stage.

Reply:
Actually this might not work with the way we have things currently build.

It is unlikely that someone would use things this way, but if you have inheritance like s3 -> remote -> s3 you could end up with all the fields set, but need to ignore some of them.

There may be ways for us to ignore this but I can't think of any that would be non-breaking.